libs/ml/include/ml/dataloaders/tensor_dataloader.hpp

```diff
@@ -42,24 +42,29 @@ class TensorDataLoader : public DataLoader<LabelType, InputType>
 public:
   TensorDataLoader() = default;
   TensorDataLoader(SizeVector const &label_shape, std::vector<SizeVector> const &data_shapes,
-                   bool random_mode = false);
+                   bool random_mode = false, float VALIDATION_TO_TRAIN_RATIO = 0.0);
```

Comment:
now I think about it, maybe both of these parameters (random & validation ratio) should instead be part of AddData?

what do you think?

Reply:
I am not convinced, maybe we can add a setter for that.